### PR TITLE
Coala: Add `Coala` plugin

### LINKED ISSUE #8: `Coala` detection

### PR DESCRIPTION
**Pull request by meetmangukiya**
Closes #8

### CODE REVIEW

File: plugins/Coala/Coala.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    def callback_message(self, msg):
+        """
+        Shout at the user for using an upercased c in coala.
+        """
+
+        if "Coala" in msg.body:
+            self.send(msg.frm,
+                      "What, what, did someone say `Coala`? @{} you're on the blacklist, "
+                      "you'll tracked down until you write a bear for `coala`. "
+                      "And remember, `coala` is ALWAYS written in lowercase `c`.!!".format(str(msg.frm).split('@')[0]))
```
Comment (meetmangukiya) [inline plugins/Coala/Coala.py]: We can edit the message to whatever we'd want, ofcourse.

Comment (nemani) [reply plugins/Coala/Coala.py]: The original bot doesn't work on `Coala` it works on a regex. That takes any spelling of coala, eg: CoaLa

File: plugins/Coala/Coala.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+class Coala(BotPlugin):
+
+    def callback_message(self, msg):
+        """
+        Shout at the user for using an upercased c in coala.
+        """
+        emots = [":(", ":angry:", ":anger:", ":confounded:",
+                 ":disappointed:", ":triumph:"]
+        
+        if re.search(r'C+[oO]+[aA]+[lL]+[aA]+', msg.body):
```
Comment (meetmangukiya) [inline plugins/Coala/Coala.py]: @nemaniarjun changed :point_up:

Comment (meetmangukiya) [reply plugins/Coala/Coala.py]: Tested at :point_up: [January 7, 2017 12:41 PM](https://gitter.im/coala/cobot-test?at=587094b0af6b364a293e3e1f)

Comment (nemani) [reply plugins/Coala/Coala.py]: The c also.

Comment (meetmangukiya) [reply plugins/Coala/Coala.py]: We don't want to respond on small `c` :/ ? This is the regex used currently in cobot

Comment (nemani) [reply plugins/Coala/Coala.py]: what if someone say "cOALA" ?

Comment (nemani) [reply plugins/Coala/Coala.py]: lets add a small c and check if the `pattern != coala`

Comment (meetmangukiya) [reply plugins/Coala/Coala.py]: could do that

File: plugins/Coala/Coala.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+class Coala(BotPlugin):
+
+    def callback_message(self, msg):
+        """
+        Shout at the user for using an upercased c in coala.
+        """
+        emots = [":(", ":angry:", ":anger:", ":confounded:",
+                 ":disappointed:", ":triumph:"]
+        
+        if re.search(r'C+[oO]+[aA]+[lL]+[aA]+', msg.body) and not re.search(r'c+o+a+l+a+'):
```
Comment (meetmangukiya) [inline plugins/Coala/Coala.py]: Split lines

File: plugins/Coala/Coala.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+class Coala(BotPlugin):
+
+    def callback_message(self, msg):
+        """
+        Shout at the user for using an upercased c in coala.
+        """
+        emots = [":(", ":angry:", ":anger:", ":confounded:",
+                 ":disappointed:", ":triumph:"]
+        
+        if re.search(r'C+[oO]+[aA]+[lL]+[aA]+', msg.body)
```
Comment (nemani) [inline plugins/Coala/Coala.py]: How about 
```
x = re.search(r'[cC]+[oO]+[aA]+[lL]+[aA]+', msg.body) 
if x and x!="coala":
    warn user of death
```

Comment (meetmangukiya) [reply plugins/Coala/Coala.py]: that'll scare newcomers away! :P

Comment (nemani) [reply plugins/Coala/Coala.py]: Just an example :P we can add it to #15

**Comment by nemani** [comment]
Merging.

**Comment by nemani** [comment]
@rultor merge

**Pull request merged**